Apps/WebClient/src/Server/Services/UserProfileService.cs

```diff
@@ -185,7 +185,7 @@ public async Task<RequestResult<UserProfileModel>> CreateUserProfile(CreateUserR
 
                 if (!string.IsNullOrWhiteSpace(requestedEmail))
                 {
-                    this.emailQueueService.QueueNewInviteEmail(hdid, requestedEmail, hostUri);
+                    this.emailQueueService.QueueNewInviteEmail(hdid, requestedEmail, hostUri, this.configurationService.GetConfiguration().WebClient.EmailVerificationExpiryHours);
```

Comment:
Lets get the configuration value on the constructor instead of here

Reply:
also this is where you can have a default value

Reply:
> Lets get the configuration value on the constructor instead of here
I'm not sure about this though, as passing in parameter in clear like this is aligned with principles of Functional Programming, passing it via constructor will hide the param inside but I'm ok if the team wants to go with constructor.

Reply:
The variable would be defined as part of the class implementation and set on the constructor. That would not hide it since is not set as global.

Reply:
I'm not sure about the pros of setting it in the constructor here though, passing by param to the function will make it clear that queue new email, but it expires in X hours, developer doesn't have to guess what this method is about and what details it requires processing, it is the principle of Functional Programming.

Reply:
The difference is that that value is constant at the time the constructor is called and will not change at any time during the existence of the class instance.  On the other hand, getting the configuration every time for this method is calling extra functions without adding anything.
Also, the method call is more readable without the nesting of functions